python/tskit/trees.py

```diff
@@ -4549,6 +4549,62 @@ def subset(self, nodes, record_provenance=True):
         tables.subset(nodes, record_provenance)
         return tables.tree_sequence()
 
+    def union(
+        self,
+        other,
+        node_mapping,
+        check_shared_equality=True,
+        add_populations=True,
+        record_provenance=True,
+    ):
+        """
+        Returns an expanded tree sequence which contains the node-wise union of
+        ``self`` and ``other``. The operation adds the non-shared portions of
+        ``other`` onto a copy of ``self``. The indices in the ``node_mapping``
+        array correspond to node ids in ``other``, and the elements encode the
+        equivalent node id in ``self`` or ``tskit.NULL`` if the node is
+        exclusive to ``other``. Nodes that are exclusive ``other`` are added to
+        a copy of ``self``, along with:
+
+        1. Individuals which are new to ``self``.
+        2. Edges whose parent or child are new to ``self``.
+        3. Sites which were not present in ``self``.
+        4. Mutations whose nodes are new to ``self``.
+
+        Nodes that are added onto ``self`` are consistently assigned new
+        populations. You may disable this behavior by setting
+        ``add_populations=False``.
+
```

Comment:
I would say this next sentence "Note that this operation also sorts the resulting tables, so the resulting tree sequence may not be equal to ``self`` even if nothing new was added."

No need to bother people about "deduplicates sites and computes mutation parents", since that's under the hood.